### PR TITLE
[Serializer] Deprecate CsvEncoder as_collection false default value

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | yes 
| Tests pass?   | yes    
| Fixed tickets | N/A   
| License       | MIT
| Doc PR        | N/A

As already expressed in https://github.com/symfony/symfony/pull/25369 and related issues, this behavior is quite counter-intuitive. It may be fine for write-API with a single document in the body but I think such CSV APIs are way less common than file-based ones, expecting collections. So I think this behavior should be opt-in explicitly in required cases, always dealing with collections by default.
This is still an arbitrary decision, but trying to make it based on use-cases and user's experience with CSV.

Note: perhaps we could find a better name for this as the semantic of setting `as_collection` to `false` to get the single row directly would not be really obvious. 
Also, it could throw an exception when getting multiple rows where only one was expected.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(rebase needed)

**Review by dunglas — APPROVED** [review]

**Comment by ogizanagi** [comment]
Rebased

**Review by chalasr — APPROVED** [review]

**Review by fabpot — DISMISSED** [review]

File: UPGRADE-4.2.md [hunk 1]
```diff
@@ -31,6 +31,12 @@ Config
 
  * Deprecated constructing a `TreeBuilder` without passing root node information.
 
+DoctrineBridge
+--------------
+
+ * The `lazy` attribute on `doctrine.event_listener` tags was removed.
+   Listeners are now lazy by default. So any `lazy` attributes can safely be removed from those tags.
```
[the remark this answers is not in the document]
Comment (ogizanagi) [reply UPGRADE-4.2.md]: Yes, indeed. It's just to respect the alpha order we use on UPGRADE files.
I can do another PR if you want, but IIRC we already accepted such changes inside unrelated PRs in the past.

[the remark this answers is not in the document]
Comment (ogizanagi) [reply UPGRADE-4.2.md]: See https://github.com/symfony/symfony/pull/27781

**Review by fabpot — APPROVED** [review]

**Review by sroze — APPROVED** [review]

**Pull request merged**